### DISCUSSION TITLE
Containerized server for development and testing

### DISCUSSION THREAD

**Discussion by gitToad** [eidolonFIRE/xcNav_web-DEPRECATED-#11] [Ideas]
Currently the app - regardless of where it runs, my phone, calebs desktop etc. always talks to a single server (snacknack.com/w/ppg/api/api.php is the API endpoint) 

While developing code (server code for sure but also client code since it affects the database state) you really want an isolated independent server that you can run against. 

Proposal is to have an easily installable (docker?) container preconfigured with PHP, Apache and MySQL  (pretty standard stuff) plus some config script in git that loads up the DB with the right schema and perhaps some well known state to start out from (and can easily return to) 

The container could run on Macs, PCs, Linux even a little raspberry pi if you want persistence and multiple clients being able to hit it.

The build system (courtesy of Caleb) would at the end of a compile/build copy all the necessary files over to the containers apache root directory.

**Comment by adrienbernede** [comment]
Since the stack is pretty standard, a quick google search lead me to this:
https://www.cloudreach.com/en/technical-blog/containerize-this-how-to-use-php-apache-mysql-within-docker-containers/

It seems pretty straightforward. I am guessing you would be interesting in how to integrate this to your development environment?

Is the server-side code already hosted somewhere? (and version-controlled?).
